msbuild/Xamarin.MacDev.Tasks.Core/Tasks/IBToolTaskBase.cs

```diff
@@ -315,54 +235,242 @@ public override bool Execute ()
 
 					LogWarningsAndErrors (dict, item);
 				} catch (Exception ex) {
-					Log.LogError ("Failed to load output manifest for {0}: {1}", ToolName, ex.Message);
+					Log.LogError ("Failed to load output log file for {0}: {1}", ToolName, ex.Message);
 					if (File.Exists (manifest.ItemSpec))
-						Log.LogError ("Output manifest contents: {0}", File.ReadAllText (manifest.ItemSpec));
+						Log.LogError ("ibtool log: {0}", File.ReadAllText (manifest.ItemSpec));
 					continue;
 				}
 
-				if (UseCompilationDirectory)
-					compiled.AddRange (GetCompilationDirectoryOutput (expected));
+				if (UseCompilationDirectory) {
+					// Note: When using a compilation-directory, we'll scan dir the baseOutputDir later as
+					// an optimization to collect all of the compiled output in one fell swoop.
+					var metadata = expected.CloneCustomMetadata ();
+
+					foreach (var target in targets)
+						mapping.Add (name + "~" + target + extension, metadata);
+
+					mapping.Add (path, metadata);
+				} else {
+					compiled.AddRange (GetCompilationOutput (expected));
+				}
+
+				manifests.Add (manifest);
+			}
+
+			if (UseCompilationDirectory)
+				compiled.AddRange (GetCompilationDirectoryOutput (baseOutputDir, mapping));
+
+			return !Log.HasLoggedErrors;
+		}
+
+		bool LinkStoryboards (string baseManifestDir, string baseOutputDir, List<ITaskItem> storyboards, List<ITaskItem> linked, IList<ITaskItem> manifests, bool changed)
+		{
+			var manifest = new TaskItem (Path.Combine (baseManifestDir, "link"));
+			var mapping = new Dictionary<string, IDictionary> ();
+			var unique = new HashSet<string> ();
+			var items = new List<ITaskItem> ();
+
+			// Make sure that `Main.storyboardc` is listed *before* `Main~ipad.storyboardc` and `Main~iphone.storyboardc`,
+			// this is important for the next step to filter out the device-specific storyboards based on the same source.
+			storyboards.Sort ((x, y) => string.Compare (x.ItemSpec, y.ItemSpec, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
The `Main` part of the filename should match afaik, but the `~ipad` portion could be `~iPad` sometimes (I'm not sure what causes that but that was a bug I recently fixed that lead to this refactoring).

That said, since the "prefix" of the filenames is what matters for sorting, we might be able to safely use `Ordinal` or at worst, `OrdinalIgnoreCase`?